### PR TITLE
fix(labrinth): ensure versions get removed from search indexes before ending route execution

### PR DESCRIPTION
**Pull request by AlexTMjugador**
These changes ensure that Labrinth waits for Meilisearch document removal requests to complete before returning from the relevant project and version edit/removal routes. This resolves the flakiness of the `index_swaps` test observed in CI, and addresses potential edge cases where a deleted project version might still be searchable for a short time after the API had already responded.

Additionally, I've refactored the `project_edit` route handler to reduce excessive block nesting by adopting clearer and more idiomatic early returns. I also moved the Meilisearch document removal step to occur alongside the Redis cache cleanup, outside of the database transaction, which should help avoiding tying up costly DB connections even if Meilisearch performance is degraded.

### CODE REVIEW

File: apps/labrinth/src/search/indexing/mod.rs [hunk 1]
```diff
@@ -41,12 +46,34 @@ pub async fn remove_documents(
...
     indexes.append(&mut indexes_next);
 
-    for index in indexes {
-        index
-            .delete_documents(
-                &ids.iter().map(|x| to_base62(x.0)).collect::<Vec<_>>(),
-            )
-            .await?;
+    let client = config.make_client()?;
+    let client = &client;
```
[the remark this answers is not in the document]
Comment (AlexTMjugador) [reply apps/labrinth/src/search/indexing/mod.rs:50]: It's not redundant, it's purposeful shadowing of a stack variable with a borrow to it to avoid the need to clone it on the `async move` block below.

**Review by Gaming32 — APPROVED** [review]
